### ISSUE TITLE
Add integration dialog is not keyboard accessible

### ISSUE THREAD

**Issue by Misiu** [home-assistant/frontend#23825]
### Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

When you go to the integration page and click on `add integration` your cursor is set in the input box, you can filter the list of integrations by typing part of the name, but you can't use your keyboard to navigate on the list.
You can't cancel by pressing Esc.
When you hit Enter the first integration on the list is picked, but it isn't highlighted.

### Describe the behavior you expected

We should be able to use the keyboard to navigate up and down the list, use Enter to continue with the selected integration or use Esc to cancel the process.

### Steps to reproduce the issue

1. Go to the integration page
2. Click on add integration
3. Type `go` ( you should see Google as first, Govee as second)
4. Try to pick `Govee` using your keyboard
5. After pressing Enter the first integration is picked (it isn't highlighted)
6. After pressing Esc the dialog isn't closed
...

### What version of Home Assistant Core has the issue?

2025.1.3

### In which browser are you experiencing the issue?

Google Chrome 131.0.6778.266

### Which operating system are you using to run this browser?

Windows 10

### State of relevant entities

```txt

```

### Problem-relevant frontend configuration

```yaml

```

### Javascript errors shown in your browser console/inspector

```txt

```
